### PR TITLE
feat: Add EMR runtime for flink operator blueprint

### PR DESCRIPTION
**Pull request by mithun008**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

Added blue print to support EMR Flink operator on EKS.

### Motivation

EMR Flink operator enhances the open source Flink operator by addition AWS specific capabilities for scaling, logging and monitoring.

### More

- [ x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ x] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ x] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [ x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
+    state.savepoints.dir: s3://emr-flink-data/savepoints
+
+
+
+  flinkVersion: v1_17
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml:27]: `emr-7.0.0` should use `v1_18`.

File: streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  # add service account
+  # serviceAccount: flink-team-a-sa
+
+  executionRoleArn: arn:aws:iam::681921237057:role/emr-eks-flink-flink-team-a-20240406012025932700000008
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml:31]: Who's this `681921237057`? Shall we dynamically set it up for customer acct?

File: streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  name: basic-example
+  namespace: flink-team-a-ns
+spec:
+  image: flink:1.16
+  flinkVersion: v1_16
+  flinkConfiguration:
+    taskmanager.numberOfTaskSlots: "2"
+    state.savepoints.dir: file:///flink/data/checkpoint/savepoints
+    state.checkpoints.dir: file:///flink/data/checkpoint/checkpoints
+    high-availability: org.apache.flink.kubernetes.highavailability.KubernetesHaServicesFactory
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml:14]: Deprecated config. Use `high-availability.type: kubernetes` instead.

File: streaming/emr-eks-flink/helm-values/aws-for-fluentbit-values.yaml
Comment (mengfwanaws) [inline streaming/emr-eks-flink/helm-values/aws-for-fluentbit-values.yaml:1]: We don't use fluentbit. Please refer to monitoring configuration instead.

File: website/docs/blueprints/streaming-platforms/emr-eks-flink.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+3. Triggering and managing savepoints
+4. Handling errors, rolling-back broken upgrades
+
+In addition to the above features, EMR Flink Kubernetes operator provides the following additional capabilities:
+1. Launching Flink application using jars in Amazon S3
+2. Monitoring integration with Amazon S3 and Amazon CloudWatch and container log rotation.
+3. Automatically tunes Autoscaler configurations based on historical trends of observed metrics.
+4. Faster Flink Job Restart during scaling or Failure Recovery
+5. IRSA (IAM Roles for Service Accounts) Native Integration
+6. Apache Airflow Integration
```
Comment (mengfwanaws) [inline website/docs/blueprints/streaming-platforms/emr-eks-flink.md:43]: I don't see an example in this PR.

File: streaming/emr-eks-flink/addons.tf [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  aws_cloudwatch_metrics = {
+    values = [templatefile("${path.module}/helm-values/aws-cloudwatch-metrics-values.yaml", {})]
+  }
+  #---------------------------------------
+  # Adding AWS Load Balancer Controller
+  #---------------------------------------
+  enable_aws_load_balancer_controller = true
+
+  #---------------------------------------
+  # AWS for FluentBit - DaemonSet
```
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply streaming/emr-eks-flink/addons.tf:135]: You are absolutely right @mengfwanaws! We use FluentBit to capture the logs from other k8s add-ons and apps running on the EKS cluster.

File: streaming/emr-eks-flink/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
+<!-- BEGIN_TF_DOCS -->
+## Requirements
+
+| Name | Version |
+|------|---------|
+| <a name="requirement_terraform"></a> [terraform](#requirement\_terraform) | >= 1.0.0 |
+| <a name="requirement_aws"></a> [aws](#requirement\_aws) | >= 3.72 |
+| <a name="requirement_helm"></a> [helm](#requirement\_helm) | >= 2.4.1 |
```
[the remark this answers is not in the document]
Comment (vara-bonthu) [reply streaming/emr-eks-flink/README.md:8]: This is the Terraform provider version and the latest version is 2.13.0 https://registry.terraform.io/providers/hashicorp/helm/latest/docs/guides/v2-upgrade-guide and it uses latest Helm version 3.13.2 https://github.com/hashicorp/terraform-provider-helm/blob/8d8f0503c39f9fb2d718805e5a619ef56728271b/go.mod#L13

@mithun008 Upgrade this to 2.13 in `versions.tf` file and rerun it

File: streaming/emr-eks-flink/addons.tf [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+            values: ["spot","on-demand"]
+          - key: "kubernetes.io/arch"
+            operator: In
+            values: ["amd64"]
+          - key: "karpenter.k8s.aws/instance-category"
+            operator: In
+            values: ["c"]
+          - key: "karpenter.k8s.aws/instance-family"
+            operator: In
+            values: ["c5d"]
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/addons.tf:189]: All our examples only recommend using c5d? Should we mix the instance types here? 

@mengfwanaws what's your view?

@mithun008 Lets add another nodepool for Graviton and we can show the example for using Graviton as well. @senkinnar FYI

File: streaming/emr-eks-flink/data.tf [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  name = module.eks.cluster_name
+}
+
+
+data "aws_ecrpublic_authorization_token" "token" {
+  provider = aws.ecr
+}
+
+data "aws_availability_zones" "available" {}
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/data.tf:12]: remove these extra lines

File: streaming/emr-eks-flink/data.tf [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+
+data "aws_caller_identity" "current" {}
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/data.tf:16]: remove these extra lines

File: streaming/emr-eks-flink/data.tf [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    name   = "name"
+    values = ["amazon-eks-node-${module.eks.cluster_version}-*"] # Update this for ARM ["amazon-eks-arm64-node-${module.eks.cluster_version}-*"]
+  }
+}
+
+
+#---------------------------------------------------------------
+# Example IAM policy for Flink job execution
+#---------------------------------------------------------------
+data "aws_iam_policy_document" "flink_operator" {
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/data.tf:34]: If this policy is created for running Flink Jobs, then don't call it as `flink_operator` for this resource. Just call it as `flink_sample_job`

File: streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    kubernetes.operator.job.autoscaler.stabilization.interval: "5s"
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
+    state.savepoints.dir: s3://emr-flink-data/savepoints
+
+
+  executionRoleArn: arn:aws:iam::xxxxxxxxx:role/emr-eks-flink-flink-team-a-20240406012025932700000008
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml:30]: We need to provide instruction to replace this execution role ARN on top of the file as a NOTE

File: streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    # Autoscaler parameters
+    kubernetes.operator.job.autoscaler.enabled: "true"
+    kubernetes.operator.job.autoscaler.scaling.enabled: "true"
+    kubernetes.operator.job.autoscaler.stabilization.interval: "5s"
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
+    state.savepoints.dir: s3://emr-flink-data/savepoints
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml:27]: This should be replaced with the S3 bucket output created by this terraform deployment. You can write these instructions on the top of the file as a NOTE.

File: streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+
+  executionRoleArn: arn:aws:iam::xxxxxxxxx:role/emr-eks-flink-flink-team-a-20240406012025932700000008
+  jobManager:
+    # Replace with s3 bucket in your own account
+    storageDir: s3://emr-flink-data/data/basic-example-app-cluster/jobmanager
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml:33]: Same as above comments

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+    # Autoscaler parameters
+    kubernetes.operator.job.autoscaler.enabled: "true"
+    kubernetes.operator.job.autoscaler.scaling.enabled: "true"
+    kubernetes.operator.job.autoscaler.stabilization.interval: "5s"
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
+    state.savepoints.dir: s3://emr-flink-data/savepoints
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:27]: refer to the same comments as above

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+
+  executionRoleArn: arn:aws:iam::xxxxxxxxx:role/emr-eks-flink-flink-team-a-20240406012025932700000008
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:30]: refer to the same comments as above

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+    metadata:
+      name: pod-template
+    spec:
+      nodeSelector:
+        NodeGroupType: "FlinkComputeOptimized"
+
+
+  jobManager:
+    # Replace with s3 bucket in your own account
+    storageDir: s3://emr-flink-data/data/basic-example-app-cluster/jobmanager
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:44]: refer to the same comments as above

File: streaming/emr-eks-flink/flink-team.tf [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+  }
+}
+
+#---------------------------------------------------------------
+# Creates IAM policy for IRSA. Provides IAM permissions for flink pods
+#---------------------------------------------------------------
+resource "aws_iam_policy" "flink" {
+  description = "IAM role policy for flink Job execution"
+  name        = "${local.name}-flink-irsa"
+  policy      = data.aws_iam_policy_document.flink_operator.json
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/flink-team.tf:17]: move the policy from `data.tf` file to this file

File: streaming/emr-eks-flink/flink-team.tf [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+      service_account = "emr-containers-sa-flink-operator"
+    }
+  }
+
+}
+
+#create a log group
+resource "aws_cloudwatch_log_group" "flink_team_a" {
+  name              = "/aws/emr-flink/flink-team-a"
+  retention_in_days = 30
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/flink-team.tf:82]: change this to 7 days and whatever is minimum

File: streaming/emr-eks-flink/locals.tf [hunk 19]
```diff
@@ -0,0 +1,13 @@
...
+
+  region = var.region
+
+  tags = {
+    Blueprint  = local.name
+    GithubRepo = "github.com/awslabs/data-on-eks"
+  }
+  flink_team     = "flink-team-a"
+  flink_operator = "flink-kubernetes-operator"
+}
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/locals.tf:13]: remove this file and move the content to `main.tf` file

File: streaming/emr-eks-flink/data.tf [hunk 20]
```diff
@@ -0,0 +1,64 @@
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/data.tf:1]: remove this file and move the content to `main.tf` file

File: streaming/emr-eks-flink/data.tf (hunk 10 above)
Comment (vara-bonthu) [reply streaming/emr-eks-flink/data.tf:34]: move this policy to `flink-team.tf`

File: streaming/emr-eks-flink/main.tf [hunk 21]
```diff
@@ -0,0 +1,222 @@
...
+      username = "system:node:{{EC2PrivateDNSName}}"
+      groups = [
+        "system:bootstrappers",
+        "system:nodes",
+      ]
+    }
+  ]
+
+
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/main.tf:30]: remove extra lines

File: streaming/emr-eks-flink/main.tf [hunk 22]
```diff
@@ -0,0 +1,222 @@
...
+        WorkerType    = "ON_DEMAND"
+        NodeGroupType = "core"
+      }
+
+      tags = {
+        Name                     = "core-node-grp",
+        "karpenter.sh/discovery" = local.name
+      }
+    }
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/main.tf:143]: remove extra lines

File: streaming/emr-eks-flink/main.tf [hunk 23]
```diff
@@ -0,0 +1,222 @@
...
+
+  }
+
+}
+
+#import module vpc
+module "vpc" {
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/main.tf:150]: move this to `vpc.tf` file

File: streaming/emr-eks-flink/main.tf [hunk 24]
```diff
@@ -0,0 +1,222 @@
...
+    "kubernetes.io/cluster/${local.name}" = "shared"
+    "kubernetes.io/role/internal-elb"     = 1
+  }
+}
+
+
+
+
+# deploy a helm chart for flink-kubernetes-operator
+resource "helm_release" "flink_kubernetes_operator" {
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/main.tf:181]: remove this and use this add-on from EKS Data addons https://github.com/aws-ia/terraform-aws-eks-data-addons/blob/main/emr-flink-operator.tf. Update the addon in this repo if required

File: streaming/emr-eks-flink/outputs.tf [hunk 25]
```diff
@@ -0,0 +1,9 @@
+# create output for flink operator role arn
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/outputs.tf:1]: add this

```
################################################################################
# EKS Managed Node Group
################################################################################

output "configure_kubectl" {
  description = "Configure kubectl: make sure you're logged in with the correct AWS profile and run the following command to update your kubeconfig"
  value       = "aws eks --region ${local.region} update-kubeconfig --name ${module.eks.cluster_name}"
}
```

File: streaming/emr-eks-flink/providers.tf [hunk 26]
```diff
@@ -0,0 +1,36 @@
+
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/providers.tf:1]: remove this file and move these providers to `main.tf`

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the PR, @mithun008! I've suggested some changes. 

Overall, I would like you to align the blueprint more closely with `emr-eks-karpenter`. Additionally, we should include more examples, which can be added in subsequent PRs.

File: streaming/emr-eks-flink/addons.tf [hunk 27]
```diff
@@ -143,11 +139,27 @@ module "eks_blueprints_addons" {
 # Data on EKS Kubernetes Addons
 #---------------------------------------------------------------
 module "eks_data_addons" {
+
+  depends_on = [module.flink_irsa_jobs, module.flink_irsa_operator]
+  #source  = "git@github.com:mithun008/terraform-aws-eks-data-addons/"
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/addons.tf:144]: Please remove the comment.

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 28]
```diff
@@ -26,8 +26,8 @@ spec:
     state.checkpoints.dir: s3://emr-flink-data/checkpoints
     state.savepoints.dir: s3://emr-flink-data/savepoints
 
-
-  executionRoleArn: arn:aws:iam::xxxxxxxxx:role/emr-eks-flink-flink-team-a-20240406012025932700000008
+  # Replace this execution role ARN with your own
+  executionRoleArn: arn:aws:iam::681921237057:role/emr-eks-flink-flink-team-a-20240406012025932700000008
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:30]: We should substitute this with variables, or provide instructions on how to generate the role arn (including the permissions).

File: streaming/emr-eks-flink/examples/karpenter/graviton-example-app-cluster.yaml [hunk 29]
```diff
@@ -0,0 +1,72 @@
...
+    kubernetes.operator.job.autoscaler.stabilization.interval: "5s"
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
+    state.savepoints.dir: s3://emr-flink-data/savepoints
+
+  # Replace this execution role ARN with your own
+  executionRoleArn: arn:aws:iam::681921237057:role/emr-eks-flink-flink-team-a-20240406012025932700000008
```
Comment (mengfwanaws) [inline streaming/emr-eks-flink/examples/karpenter/graviton-example-app-cluster.yaml:30]: Same.

File: streaming/emr-eks-flink/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (mithun008) [reply streaming/emr-eks-flink/README.md]: done

File: streaming/emr-eks-flink/addons.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (mithun008) [reply streaming/emr-eks-flink/addons.tf:131]: done

File: streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml (hunk 1 above)
Comment (mithun008) [reply streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml]: done

File: streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml (hunk 2 above)
Comment (mithun008) [reply streaming/emr-eks-flink/examples/basic-example-app-cluster.yaml]: removed it and added comments as well as docs.

File: website/docs/blueprints/streaming-platforms/emr-eks-flink.md (hunk 4 above)
Comment (mithun008) [reply website/docs/blueprints/streaming-platforms/emr-eks-flink.md]: removed

File: streaming/emr-eks-flink/addons.tf (hunk 7 above)
Comment (mithun008) [reply streaming/emr-eks-flink/addons.tf:198]: done

File: streaming/emr-eks-flink/data.tf (hunk 8 above)
Comment (mithun008) [reply streaming/emr-eks-flink/data.tf]: done

File: streaming/emr-eks-flink/data.tf (hunk 9 above)
Comment (mithun008) [reply streaming/emr-eks-flink/data.tf]: done

File: streaming/emr-eks-flink/data.tf [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+  owners      = ["amazon"]
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["amazon-eks-node-${module.eks.cluster_version}-*"] # Update this for ARM ["amazon-eks-arm64-node-${module.eks.cluster_version}-*"]
+  }
+}
+
+
```
[the remark this answers is not in the document]
Comment (mithun008) [reply streaming/emr-eks-flink/data.tf]: done

**Comment by mithun008** [comment]
@vara-bonthu I have addressed your comments. Thank you!

File: streaming/emr-eks-flink/outputs.tf [hunk 31]
```diff
@@ -0,0 +1,27 @@
...
+output "flink_jobs_role_arn" {
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/outputs.tf:2]: flink_job_execution_role_arn

File: streaming/emr-eks-flink/outputs.tf [hunk 32]
```diff
@@ -0,0 +1,27 @@
...
+}
+output "flink_jobmanager_path" {
+  value       = "s3://${module.s3_bucket.s3_bucket_id}/jobmanager"
+  description = "S3 path for jobmanager data"
+}
+
+output "flink_logs_path" {
+  value       = "s3://${module.s3_bucket.s3_bucket_id}/logs"
+  description = "S3 path for logs"
+}
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/outputs.tf:27]: remove all of these and add only s3bucet id

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 33]
```diff
@@ -0,0 +1,72 @@
...
+
+    # Autoscaler parameters
+    kubernetes.operator.job.autoscaler.enabled: "true"
+    kubernetes.operator.job.autoscaler.scaling.enabled: "true"
+    kubernetes.operator.job.autoscaler.stabilization.interval: "5s"
+    kubernetes.operator.job.autoscaler.metrics.window: "1m"
+
+    jobmanager.scheduler: adaptive
+    # Replace with s3 bucket in your own account
+    state.checkpoints.dir: s3://emr-flink-data/checkpoints
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:26]: <<REPLACE_WITH_S3_BUCKET_ID>>

File: streaming/emr-eks-flink/versions.tf [hunk 34]
```diff
@@ -0,0 +1,33 @@
...
+      source  = "hashicorp/random"
+      version = "3.3.2"
+    }
+  }
+
+  # ##  Used for end-to-end testing on project; update to suit your needs
+  # backend "s3" {
+  #   bucket = "doeks-github-actions-e2e-test-state"
+  #   region = "us-west-2"
+  #   key    = "e2e/flink/terraform.tfstate"
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/versions.tf:31]: emr-eks-flink

File: streaming/emr-eks-flink/addons.tf [hunk 35]
```diff
@@ -0,0 +1,247 @@
...
+  version           = "~> 1.30" # ensure to update this to the latest/desired version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  #---------------------------------------------------------------
+  # EMR Flink operator
+  #---------------------------------------------------------------
+  enable_emr_flink_operator = true
+  emr_flink_operator_helm_config = {
+    namespace                = "emr-flink-operator"
+    create_namespace         = true
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/addons.tf:124]: remove this line as this is a default

File: streaming/emr-eks-flink/addons.tf [hunk 36]
```diff
@@ -0,0 +1,247 @@
...
+    namespace                = "${local.flink_operator}-ns"
+    name                     = "flink-kubernetes-operator"
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/addons.tf:126]: remove this line as this is default

File: streaming/emr-eks-flink/addons.tf [hunk 37]
```diff
@@ -0,0 +1,247 @@
...
+    repository               = "oci://public.ecr.aws/emr-on-eks"
+    chart                    = "flink-kubernetes-operator"
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/addons.tf:128]: remove this line as this is default

File: streaming/emr-eks-flink/addons.tf [hunk 38]
```diff
@@ -0,0 +1,247 @@
...
+
+  source            = "aws-ia/eks-data-addons/aws"
+  version           = "~> 1.30" # ensure to update this to the latest/desired version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+
+  #---------------------------------------------------------------
+  # EMR Flink operator
+  #---------------------------------------------------------------
+  enable_emr_flink_operator = true
+  emr_flink_operator_helm_config = {
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/addons.tf:122]: replace this bloc with 

```tf
  emr_flink_operator_helm_config = {
    repository               = "oci://public.ecr.aws/emr-on-eks"
    operatorExecutionRoleArn = module.flink_irsa_operator.iam_role_arn
  }
  
```

You can remove the `repository` and `operatorExecutionRoleArn` once you update these in Data addons repo

File: streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml [hunk 39]
```diff
@@ -0,0 +1,64 @@
+# NOTE: Make sure you you replce <ENTER_S3_BUCKET> with your flink_operator_bucket output before running this job
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/cluster-autoscaler/flink-sample-job.yaml:1]: check spelling and repeated words

File: streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml [hunk 40]
```diff
@@ -0,0 +1,74 @@
+# NOTE: Make sure you you replce <ENTER_S3_BUCKET> and with your S3 Bucket before running this job.
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/basic-example-app-cluster.yaml:1]: check spelling and repeated words

File: streaming/emr-eks-flink/examples/karpenter/graviton-example-app-cluster.yaml [hunk 41]
```diff
@@ -0,0 +1,75 @@
+# NOTE: Make sure you you replce <ENTER_S3_BUCKET> with your flink_operator_bucket output before running this job
```
Comment (vara-bonthu) [inline streaming/emr-eks-flink/examples/karpenter/graviton-example-app-cluster.yaml:1]: check spelling and repeated words

**Review by vara-bonthu — COMMENTED** [review]
left few minor comments otherwise its good

**Review by vara-bonthu — APPROVED** [review]
/lgtm

**Pull request merged**